internal/errors/error.go

```diff
@@ -104,18 +139,25 @@ func (e *Err) Error() string {
 		join(&s, ": ", e.Msg)
 	}
 
-	if info, ok := errorCodeInfo[e.Code]; ok {
+	var skipInfo bool
+	var wrapped *Err
+	if As(e.Wrapped, &wrapped) {
+		// if wrapped error code is the same as this error, don't print redundant info
```

Comment:
What if it's the same code but different message/kind/op? This makes an assumption that although it's _possible_ to specify the same code when wrapping an error (since we are supporting multiple codes) that only one place in the stack will actually do so with the same error code. I don't think that's a good assumption. If people can specify codes they very well might...

Reply:
`Message` and `Op` are always printed if not empty, `Code` was specifically left out, with this check, to avoid the following when the stringer is invoked:

```
"static.CreateHost: invalid parameter: parameter violation: error #100 static.newHostId: invalid parameter: parameter violation: error #100 db.newId: missing prefix: parameter violation: error #100"
```

Instead, we get:

```
static.CreateHost: static.newHostId: db.newId: missing prefix: parameter violation: error #100
```

The `parameter violation: error #100` part, which comes from https://github.com/hashicorp/boundary/blob/main/internal/errors/info.go#L15 is printed each time we recursively print the wrapped error

Kind is a field based on code within the info map, I am not sure how we can have the same code but a different Kind?